oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -482,17 +481,52 @@ private ImpersonatedCredentials(Builder builder) {
     }
   }
 
+  /**
+   * Gets the universe domain for the credential.
+   *
+   * @return An explicit universe domain if it was explicitly provided, use the universe domain from
+   *     source credentials otherwise
+   */
+  @Override
+  public String getUniverseDomain() throws IOException {
+    if (isExplicitUniverseDomain()) {
+      return super.getUniverseDomain();
+    }
+    return this.sourceCredentials.getUniverseDomain();
+  }
+
+  /**
+   * Checks if universe domain equals to {@link Credentials#GOOGLE_DEFAULT_UNIVERSE}.
+   *
+   * @return true if universeDomain equals to {@link Credentials#GOOGLE_DEFAULT_UNIVERSE}, false
+   *     otherwise
+   */
+  @Override
+  boolean isDefaultUniverseDomain() {
+    try {
+      return getUniverseDomain().equals(Credentials.GOOGLE_DEFAULT_UNIVERSE);
+    } catch (IOException e) {
+      // super method does not throw IOException, so wrap it here.
+      // This should not happen for this credential type.
+      throw new IllegalStateException(e);
+    }
+  }
+
   @Override
   public AccessToken refreshAccessToken() throws IOException {
     if (this.sourceCredentials.getAccessToken() == null) {
       this.sourceCredentials =
           this.sourceCredentials.createScoped(Arrays.asList(CLOUD_PLATFORM_SCOPE));
     }
 
-    try {
-      this.sourceCredentials.refreshIfExpired();
-    } catch (IOException e) {
-      throw new IOException("Unable to refresh sourceCredentials", e);
+    // skip for nonGDU because it uses self-signed JWT
+    // and will get refreshed at initialize request step
```

Comment:
I'm still not sure why this call is required for GDU but not required for nonGDU, is there any feature that is only supported in GDU?

Reply:
I am convinced that this is not needed for nonGDU, reasons in description above.
For GDU, I am not 100% convinced that this is required before initialize step. But out of precautious, I do not want to change the flow for GDU in this pr. I will raise a separate issue to investigate for GDU flow.

Reply:
as comment states - nonGDU is always SSJ, no token endpoint, no refresh call

Reply:
to answer Blake's question - GDU has token endpoint, nonGDU does not :) 

SSJ is a primary flow is GDU now as well. This should be a fallback scenario if SSJ is disabled, execute conditionally if SSJ flag is disabled.

Reply:
updated condition to also skip for GDU SSJ flow.